### PR TITLE
feat(tracimcli): try to get config file location from env

### PR DESCRIPTION
**Pull request by Millefeuille42**
#6312 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Review by buxx — APPROVED** [review]
@Millefeuille42 Just think to update documentation to indicate this env var exist.

**Comment by lebouquetin** [comment]
this development has impact for sysadmin; you must validate with targeted users. I expect @PhilippeAccorsi or @yvan-algoo to validate the development and behavior, not only a developer

**Comment by Millefeuille42** [comment]
For sys-admin reviewers:
Please review `doc/backend/env_settings.md` and `doc/backend/env_settings.md`.

Purpose:
This update introduces support for the environment variable TRACIM_CONFIG__FILEPATH, which is utilized by `tracimcli` to determine the location of the configuration file. This enhancement eliminates the need to specify the configuration file location, particularly in Docker environments.

For clarity and consistency, this environment variable has been incorporated into the configuration parser. Although it's exclusively used by `tracimcli` and processed before the configuration parser loads, it's included for uniformity and cannot be read from the configuration file.

**Review by yvan-algoo — APPROVED** [review]

**Comment by PhilippeAccorsi** [comment]
Thanks for the last commit. I'm going to test with docker images.

**Comment by PhilippeAccorsi** [comment]
@Millefeuille42 small fix and I can approuve :)

File: doc/backend/setting.md [hunk 1]
```diff
@@ -22,6 +22,15 @@ You can configure Tracim's external URL with the `website.base_url` parameter, f
...
 ```
 
+## Define path to configuration file
+
+To be able to use tracimcli from any directory, you can define the path to the configuration file with the environment variable `TRACIM_CONFIG__FILEPATH`.
+Note that despite being present in the configuration spec, this variable is evaluated by the tracimcli before being parsed by the configuration parser.
+
+This means that you can use it in the env to define the path to the configuration file, but you can't use it in the configuration file to define the path to the configuration file itself.
+
+```bash
```
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply doc/backend/setting.md:30]: fixed

File: doc/backend/setting.md [hunk 2]
```diff
@@ -22,6 +22,15 @@ You can configure Tracim's external URL with the `website.base_url` parameter, f
 website.base_url = https://mysuperdomainame.ndd
 ```
 
+## Define path to configuration file
+
+To be able to use tracimcli from any directory, you can define the path to the configuration file with the environment variable `TRACIM_CONFIG__FILEPATH`.
+Note that despite being present in the configuration spec, this variable is evaluated by the tracimcli before being parsed by the configuration parser.
+
+This means that you can use it in the env to define the path to the configuration file, but you can't use it in the configuration file to define the path to the configuration file itself.
```
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply doc/backend/setting.md]: fixed

**Review by PhilippeAccorsi — APPROVED** [review]
thanks :)
Ready to merge

**Pull request merged**